compiler/rustc_middle/src/ty/fast_reject.rs

```diff
@@ -120,6 +121,7 @@ pub fn simplify_type<'tcx>(
         ty::Str => Some(SimplifiedType::Str),
         ty::Array(..) => Some(SimplifiedType::Array),
         ty::Slice(..) => Some(SimplifiedType::Slice),
+        ty::Pat(..) => Some(SimplifiedType::Pat),
```

Comment:
We should simplify pattern types like their scrutinee, so that all the pattern impls in the whole crate don't get lumped into one bucket.

Reply:
we shouldn't even have impls with pattern types. there shouldnt be any need to simplify them

Reply:
It's impossible to not have builtin impls, and rather annoying not to have user-written impls. See https://rust-lang.zulipchat.com/#narrow/stream/326866-t-types.2Fnominated/topic/Pattern.20types.20types-team.23126/near/427670099 for where I started a discussion about this

Reply:
can you get someone else to second your mcp and review this then, i dont really wanna think about the interactions of pattern types with the type system like this

Reply:
I'll review it; I've already started to do so.

I think I generally agree that not having impls is kind of unwieldy, and I'm not particularly worried about them affecting stable code or the trait solvers since they're not, e.g., subtypes of regular types.